variables.tf

```diff
@@ -4,8 +4,79 @@ variable "name" {
   type        = string
 }
 
+variable "secret_description" {
+  default     = "A password created by Terraform" #tfsec:ignore:GEN001
+  description = "Set a description for the secret"
+  type        = string
+}
+
 variable "tags" {
   default     = {}
   description = "User-Defined tags"
   type        = map(string)
 }
+
+variable "length" {
+  description = "Length of the password to be created"
+  type        = number
+}
+
+variable "lower" {
+  default     = true
+  description = "Whether to use lower case characters"
+  type        = bool
+}
+
+variable "min_lower" {
+  default     = 0
+  description = "Minimum number of lowercase letters"
+  type        = number
+}
+
+variable "number" {
+  default     = true
+  description = "Whether to use numbers"
+  type        = bool
+}
+
+variable "min_numeric" {
+  default     = 1
+  description = "Minimum number of numeric characters to use. Must be at least 1"
+  type        = number
+}
+
+variable "special" {
+  default     = true
+  description = "Whether to use special characters"
+  type        = bool
+}
+
+variable "min_special" {
+  default     = 1
+  description = "Minimum number of special characters to use. Must be at least 1"
+  type        = number
+}
+
+variable "override_special" {
+  default     = "!@#$%&*()-_=+[]{}<>:?"
+  description = "Supply your own list of special characters to use for string generation"
+  type        = string
+}
+
+variable "upper" {
+  default     = true
+  description = "Whether to use uppercase characters"
+  type        = bool
+}
+
+variable "min_upper" {
+  default     = 1
+  description = "Minimum number of uppercase characters to use. Must be at least 1"
+  type        = number
+}
+
+variable "keepers" {
```

Comment:
k is a letter, here's how to make the list of letters with python
```
>  python                                                                                                                                                                                                                                   
Python 3.8.5 (default, Jul 21 2020, 10:48:26) 
[Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import string
>>> string.ascii_lowercase
'abcdefghijklmnopqrstuvwxyz'
```

Reply:
In trying to find creative ways to point out alphabetization errors, you've made yourself look like an asshole.

Reply:
If we're going to standardize on python, you need to learn it